interfaces/builtin/steam_support.go

```diff
@@ -0,0 +1,202 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const steamSupportSummary = `allows access to various permissions required to use Steam`
+
+const steamSupportBaseDeclarationSlots = `
+  steam-support:
+    allow-installation: false
+    deny-connection: true
+    deny-auto-connection: true
+`
+
+const steamSupportConnectedPlugSecComp = `
+# Description: Allow Steam games to work correctly.
+# This extends the browser-support base seccomp. They cannot be used
+# together as browser-support will deny ptrace capabilities.
+
+# This is required for Feral Interactive ports
+ptrace
+
+# Required by titles such as Dying Light
+sched_setaffinity
+`
+
+const steamSupportConnectedPlugAppArmor = `
+# Description: Provide support for the Steam Client and Steam Games to run
+# correctly under full AppArmor confinement. Primary use case is to leverage
+# solus-runtime-gaming and linux-steam-integration to provide a sane and safe
+# Steam experience across all platforms, with appropriate mediation.
+# This AppArmor definition is used in conjunction with the browser-support
+# base AppArmor definition. They cannot both be used together as defined
+# connections however, as browser-support will deny the ptrace capabilities
+# that are required by Steam.
+
+########################
+# Steam Client / API   #
+########################
+
+# Allow Valve IPC to work for the multiprocess client architecture.
+# "Server" component will write/create/lock, clients (i.e. running games in
+# the Steam API) will mmap the object.
+owner /{dev,run}/shm/u*-ValveIPCSharedObj* mrwk,
+
+# Linux Steam Integration config mask system
+owner /{dev,run}/shm/u*-LinuxSteamIntegration.unity3d.* rw,
+
+# i.e. /dev/shm/#539043
+owner /{dev,run}/shm/#* mrw,
```

Comment:
Can this be written as `shm/#[0-9]+`?

Reply:
Update; we now know apparmor regular expressions don't grok `+` so we should use`*` instead

Reply:
:D